lib/parser.rb

```diff
@@ -12,4 +15,14 @@ def print_first_line
   def close_file
     @file.close
   end
+
+  def number_of_lines
+    @lines = @file.count 
```

Comment:
Não há necessidade de atribuir o resultado a uma variável de instância, vc pode retornar o resultado diretamente. E também, como esse método faz parte da implementação interna da classe, ele deveria ser privado.

```suggestion
    @file.count 
```